README.md

```diff
@@ -158,9 +158,49 @@ export default withDi(MyComponent, [ModalOpenDi]);
 `withDi` supports the same API and capabilities as `DiProvider`, where `target` is the third argument of the HOC `withDi(MyComponent, [Modal], MyComponent)` in case you want to limit injection to a specific component only.
 
 When you have the same dependency replaced multiple times, there are two behaviours that determine which injectable will "win":
+
 - the one defined on the closest `DiProvider` wins. So you can declare more specific replacements by wrapping components with `DiProvider` or `withDi` and those will win over same type injectables on other top level `DiProvider`s
 - the injectable defined last in the `use` array wins. So you can define common injectables but still override each type case by case (eg: `<DiProvider use={[...commonDeps, specificInjectable]}>`
 
+### Using injection replacement outside of React
+
+The usage outside React is not much different, aside from the different way of clearing the replacements.
+
+```js
+import { fetchApi } from './fetch';
+
+export async function myApiFetcher() {
+  // "mark" any type of function/class as injectable
+  di(fetchApi);
+
+  const { data } = await fetchApi();
+  return data;
+}
+```
+
+In the tests, you can use either `globalDi` directly or the wrapper `runWithDi`. The main difference is that `runWithDi` will clear the replacements for you after function execution is terminated. Such util also handles async code, but might not work effectively with scheduled code paths, or event driven implementations. In those scenarios we recommend manually adding `globalDi.clear()` to `afterEach` to ensure there is no pollution between tests.
+
+```js
+import { injectable, globalDi, runWithDi } from 'react-magnetic-di';
+import { myApiFetcher, fetchApi } from '.';
+
+it('should call the API', async () => {
+  const fetchApiDi = injectable(
+    fetchApi,
+    jest.mock().mockResolvedValue('mock')
+  );
+  // either using globalDi use/clear manually
+  globalDi.use([fetchApiDi]);
+  const result = await myApiFetcher();
+  globalDi.clear();
```

Comment:
Having `jest.mock` from one side it may be more sound to have `magnetic.use` on the other.
`globalDi` does not really resonate with the simular API.

Reply:
Hmmm... I guess, even if I find `import { jest } from '@jest/globals'` weird. 
I feel like `magnetic` as namespace makes even less sense. And `di` is already taken. If we drop the namespace we could have `diUse` and `diClear` or (`useDi` and `clearDi`)

Reply:
So basically we have 2 use cases:
- `runWithDi` - no objections to this API
- `use` + `we don't care` + `clear` - I think it will be used less often, if ever.

I have simular _different_ versions of configuration for [rewiremock](https://github.com/theKashey/rewiremock) and many years of usage defined 3 "sound" versions. I will translate them to magnetic API:
- fire and forget (99% of use cases)
```tsx
const defaultDeps = [fetchApiDi];
//
it('does something', () => {
  const result = await runWithDi(() => myApiFetcher(), [...defaultDeps, someInjectable]);
``` 
- preconfigure environment, run stuff. Like we do with `jest.mock` right now - they affect the whole file.
```tsx
globalDi.use([fetchApiDi]);
afterAll(globalDi.cleanUp);

it('does something', () => {
  const result = await runWithDi(() => myApiFetcher(), [someInjectable]);
});

it('does something else ', () => {
  // or just run, everything is preconfigured
  const result = await myOtherApiFetcher();
```

A little problem with example above is `runWithDi` will do `globalDi.clear` on exit, removing "environmental mocks".

There are two ways of solving this problem:
- hide `globalDi` for now
- add push/pull ability to mocks, so you can cleanup only added information

There could be no problem:
- as one can see `runWithDi` version is providing a little mode direct and explicit control over execution and while `globalDi` is a good addition - without care it can create more problems than resolve.

Reply:
> hide globalDi for now

Ok, let's go for keeping `globalDi` hidden for now. I feel the if someone has issues with auto-cleanup, the can always wrap the entire test in `runWithDi`